File: linechart/src/main/java/com/davidmiguel/linechart/model/Label.java
package com.davidmiguel.linechart.model;

import android.graphics.RectF;

/**
 * Encapsulates data to draw a label.
 */
public class Label {

    private RectF background;
    private float textX;
    private float textY;
    private String text;

    public Label(RectF background, float textX, float textY, String text) {
        this.background = background;
        this.textX = textX;
        this.textY = textY;
        this.text = text;
    }

    public RectF getBackground() {
        return background;
    }

    public float getTextX() {
        return textX;
    }

    public float getTextY() {
        return textY;
    }

    public String getText() {
        return text;
    }
}

File: linechart/src/main/java/com/davidmiguel/linechart/model/Label.kt
package com.davidmiguel.linechart.model

import android.graphics.Paint
import android.graphics.RectF

/**
 * Encapsulates data to draw a label.
 */
data class Label(
    val background: RectF,
    val textX: Float,
    val textY: Float,
    val text: String,
    val labelBackgroundPaint: Paint,
    val labelTextPaint: Paint,
)
